LoM-Install/arista/LoM-install.sh

```diff
@@ -0,0 +1,432 @@
+function log() {
+    local message="$1"
+    local lineno="$2"
+    local funcname="$3"
+    local filename=$(basename "$0")
+    local log_message="LOM-StartUp: $filename:$lineno: $funcname: $message"
+
+    if [[ $LOG_TO == "syslog" || $LOG_TO == "both" ]]; then
+        logger -p $facility.info "$log_message"
+    fi
+
+    if [[ $LOG_TO == "console" || $LOG_TO == "both" ]]; then
+        if [[ $LOG_TO == "console" ]]; then
+            local separator=$(printf '%0.s_' {1..30})
+            echo "$separator"
+        fi
+        echo "$log_message"
+        if [[ $LOG_TO == "console" ]]; then
+            echo "$separator"
+        fi
+    fi
+}
+
+# This function adds a key-value pair to a JSON file.
+# If the file does not exist or if its contents are not valid JSON,
+# it creates an empty dictionary and adds the key-value pair to it.
+# It then writes the dictionary back to the file in JSON format.
+# Arguments:
+#   $1: The key to add to the JSON file.
+#   $2: The value to associate with the key in the JSON file.
+#   $3: The path to the JSON file.
+function add_config() {
+    local key="$1"
+    local value="$2"
+    local file="$3"
+
+    python -c "import json
+try:
+    with open('$file', 'r') as f:
+        data = json.load(f)
+except (IOError, ValueError):
+    data = {}
+if '$value' == 'true':
+    py_value = True
+elif '$value' == 'false':
+    py_value = False
+else:
+    try:
+        py_value = int('$value')
+    except ValueError:
+        py_value = '$value'
+data['$key'] = py_value
+with open('$file', 'w') as f:
+    json.dump(data, f)"
+}
+
+# Function: install
+# Description: Installs the software.
+#   If this is the first time installation, it copies the contents from the extraction directory to the 'active' directory under /mnt/flash/lom and starts the installation. 
+#       If the installation is successful, it writes the status and existing configuration on the device to a file.
+#       If the installation is not successful, it cleans the installation, removes the active installer directory and returns False.
+#   If there is a previous installation, it cleans the previous installation, moves it to the backup directory, copies the contents from the extraction directory 
+#       to the active installer directory, and starts the installation.
+#           If the installation is successful, it writes the status and existing configuration on the device to a file.
+#           If the installation is not successful, it cleans the installation, removes the active installer directory, performs a rollback. If rollback is successful, 
+#           it returns True else False. Failure in rollback is unrecoverable and the installation returns False.
+#
+# To-Do : Goutham : We are returning False in case of installation failure and 
+# rollback failure. We need to differentiate between these two cases.
+#
+# Returns:
+# 0 if the installation was successful, 1 otherwise.
+function install() {    
```

Comment:
Add To-Do comment : Differentiate between full success install and failed install.

Reply:
Added